tests/responses/test_fileresponse.py

```diff
@@ -9,31 +10,98 @@
 from django.test import SimpleTestCase
 
 
+class UnseekableBytesIO(io.BytesIO):
+    def seekable(self):
+        return False
+
+
 class FileResponseTests(SimpleTestCase):
-    def test_file_from_disk_response(self):
+    def test_content_length_file(self):
         response = FileResponse(open(__file__, 'rb'))
         self.assertEqual(response.headers['Content-Length'], str(os.path.getsize(__file__)))
-        self.assertIn(response.headers['Content-Type'], ['text/x-python', 'text/plain'])
-        self.assertEqual(
-            response.headers['Content-Disposition'],
-            'inline; filename="test_fileresponse.py"',
-        )
         response.close()
 
-    def test_file_from_buffer_response(self):
+    def test_content_length_buffer(self):
         response = FileResponse(io.BytesIO(b'binary content'))
         self.assertEqual(response.headers['Content-Length'], '14')
+
+    def test_content_type_file(self):
+        response = FileResponse(open(__file__, 'rb'))
+        self.assertIn(response.headers['Content-Type'], ['text/x-python', 'text/plain'])
+        response.close()
+
+    def test_content_type_buffer(self):
+        response = FileResponse(io.BytesIO(b'binary content'))
         self.assertEqual(response.headers['Content-Type'], 'application/octet-stream')
+
+    def test_content_type_buffer_explicit(self):
+        response = FileResponse(io.BytesIO(b'binary content'), content_type='video/webm')
+        self.assertEqual(response.headers['Content-Type'], 'video/webm')
+
+    def test_content_type_buffer_named(self):
+        test_tuples = (
+            ('test_fileresponse.py', ['text/x-python', 'text/plain']),
+            ('test_fileresponse.pynosuchfile', ['application/octet-stream']),
+        )
+        for filename, content_types in test_tuples:
+            with self.subTest(filename=filename):
+                buffer = io.BytesIO(b'binary content')
+                buffer.name = filename
+                response = FileResponse(buffer)
+                self.assertIn(response.headers['Content-Type'], content_types)
+
+    def test_content_disposition_file(self):
+        filenames = (
+            ('', 'test_fileresponse.py'),
+            ('custom_name.py', 'custom_name.py'),
+        )
+        dispositions = (
+            (False, 'inline'),
+            (True, 'attachment'),
+        )
+        for (filename, header_filename), (as_attachment, header_disposition)\
+                in itertools.product(filenames, dispositions):
+            with self.subTest(filename=filename, disposition=header_disposition):
+                response = FileResponse(open(__file__, 'rb'), filename=filename, as_attachment=as_attachment)
+                self.assertEqual(
+                    response.headers['Content-Disposition'],
+                    '%s; filename="%s"' % (header_disposition, header_filename),
```

Comment:
oh, i love f-strings and i would've used them already.
however, in the previous PR [i was told to not use them yet](https://github.com/django/django/pull/11011#discussion_r258771475).
> Stick to .format for now. Until we settle to switch to f strings for the whole code base we should stick to format.

however again, it's been literally 2 years since then - are we there yet?

Reply:
Yes, let it rip!

Reply:
> Yes, let it rip!

Well, in moderation... 😄 But, yes, they are now accepted.

See halfway down the Python style section of the [coding style documentation](https://docs.djangoproject.com/en/3.1/internals/contributing/writing-code/coding-style/) for guidance.

Readability is key. (Personally I normally am more flexible than these rules, but I totally get the point after recently unpicking f-strings nested three or four levels deep. *Shudder.*)